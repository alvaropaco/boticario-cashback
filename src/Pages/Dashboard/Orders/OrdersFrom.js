import React, { useState } from 'react';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import Alert from '@material-ui/lab/Alert';
import InputAdornment from '@material-ui/core/InputAdornment';
import Input from '@material-ui/core/Input';
import moment from 'moment';
import { create } from '../../../services/orders';

const initialState = () => {
  const dt = new Date();
  return {
    datetime: dt.getTime()
  }
}

const OrdersForm = () => {
  const [ error, setError ] = useState(null);
  const [ token, setToken ] = useState(null);
  const [values, setValues] = useState(initialState());
  
  const onchange = async e => {
    const { value, name } = e.target;
    setValues({
      ...values,
      [name]: value
    })
  }

  const onSubmit = async e => {
    e.preventDefault();
    const { token, error } = await create(values)
    
    if(error) {
      await setError(error);
      setTimeout(() => { setError(null); }, 9000);
    }
    
    await setToken(token)
    setTimeout(() => { setToken(null); }, 9000);
  }

  return (
    <form>
      <Typography variant="h6" gutterBottom>
        Cadastrar nova Compra
      </Typography>
      <Grid container spacing={3}>
        <Grid item xs={12} md={6}>
          <TextField 
            required 
            id="code"
            name="code" 
            label="Código da Compra" 
            fullWidth
            onChange={onchange}
            value={values.code} 
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <Input
            required
            id="amount"
            name="amount"
            label="Valor total"
            fullWidth
            onChange={onchange}
            startAdornment={<InputAdornment position="start">$</InputAdornment>}
            value={values.amount}
          />
        </Grid>
        <Grid item xs={12} md={6}>
          <TextField
            id="datetime"
            name="datetime"
            label="Data"
            type="date"
            defaultValue={moment(values.datetime).format('DD/MM/YYYY')}
            fullWidth
            onChange={onchange}
            InputLabelProps={{
              shrink: true,
            }}
          />
        </Grid>
        <Grid item xs={12}>
          { 
          error 
            ? <Alert severity="error">{error}</Alert>
            : <></>
          }
          {
            token
              ? <Alert severity="success">This is a success alert — check it out!</Alert>
              : <></>
          }
          <Button variant="contained" color="primary" onClick={onSubmit}>
            Cadastrar
          </Button>
        </Grid>
      </Grid>
    </form>
  );
}

export default OrdersForm;
