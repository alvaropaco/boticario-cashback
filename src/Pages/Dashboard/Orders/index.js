import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import moment from 'moment';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Divider from '@material-ui/core/Divider';
import { loadOrders } from '../../../services/orders';

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    backgroundColor: theme.palette.background.paper,
  },
  inline: {
    display: 'inline',
  },
}));

const Orders = () => {
  const [values, setValues] = useState([])
  const classes = useStyles();

  useEffect(() => {
      fetchData();
    },
    []
  );

  const fetchData = async () => {
    const { orders } = await loadOrders()
    await setValues(orders)
  }

  function formatCurrency(numberStr){
    const number = +numberStr;
    var formatted = "R$ " + number.toFixed(2).replace(".",",");
    return formatted;
  }

  const getCashBackAmount = (amount, percentage) => {
    return formatCurrency(amount * percentage)
  }

  const getStatusLabel = statusCode => {
    const statuses = {
      0:"Em validaÃ§Ã£o", 
      1: "Reprovado", 
      2: "Aprovado"
    };
    return statuses[statusCode];
  }
  const getDateStr = timestamp => {
    return moment(Date(timestamp)).format('DD/MM/YYYY');
  }

  const renderOrdersList = orders => {
    orders = orders || []
    return (<List className={classes.root}>
        {orders.map(order => 
          <ListItem key={order.id} alignItems="flex-start">
            <Divider variant="inset" component="li" />
            <ListItemText
              primary={`${getDateStr(order.datetime)} - ${order.code} (${getStatusLabel(order.status)})`}
              secondary={`Total: R$ ${formatCurrency(order.amount)} - Cashback: R$ ${getCashBackAmount(order.amount, order.cashBackPercent)} (${order.cashBackPercent}%)`}
            />
          </ListItem>
          )}
      </List>)
  }

  return (
    <>
      <h2>Lista de Compras</h2>
      {renderOrdersList(values)}
    </>);
}

export default Orders;
