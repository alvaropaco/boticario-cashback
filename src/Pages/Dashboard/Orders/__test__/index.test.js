import React from 'react';
import Orders from '../';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom'
import { createBrowserHistory } from "history";

const history = createBrowserHistory();

it('renderiza a pÃ¡gina de Compras', () => {
    const tree = renderer
    .create(
        <Router history={history}>
            <Orders />
        </Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});