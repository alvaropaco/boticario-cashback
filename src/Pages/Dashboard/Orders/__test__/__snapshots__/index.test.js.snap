// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza a p√°gina de Compras 1`] = `
Array [
  <h2>
    Lista de Compras
  </h2>,
  <ul
    className="MuiList-root makeStyles-root-1 MuiList-padding"
  />,
]
`;
